### DISCUSSION TITLE
Building mavp2p for openWRT

### DISCUSSION THREAD

**Discussion by Autonomost** [bluenviron/mavp2p#49] [Q&A]
Hi there, 

I am trying to build mavp2p.ipk for a specific openwrt version and I keep getting compilation errors. In your readme file where should i specify my exact hardware? Running just your commands leads to a lot of issues. Any help would be appreciated please! Thanks!

**Comment by aler9** [comment]
Hello, since the ipk cross-compilation method that you mentioned is too complicated for some users, i updated the README and provided a simpler method to install on OpenWrt that doesn't involve packages or cross-compilation:

https://github.com/bluenviron/mavp2p#openwrt-binary

**Reply by Autonomost** [reply]
Thanks for the response. Unfortunately the device doesn't have enough memory to download and run git. do you have any more tips? 
Also - are their any flags you recommend to optimize the binary when its built? When i build mavp2p for this device it seems to always OOM kill the process. Any help would be fantastic please! Thanks!

**Reply by aler9** [reply]
@Autonomost i was expecting this so i added instructions on how to cross-compile the software from another machine: https://github.com/bluenviron/mavp2p#cross-compile

Regarding performance optimization, Golang always compile binaries in order for them to have the best performance.
Regarding size optimization, you can strip the debugging symbols from the executable. Search online for how to do this when compiling Go software.
